### PR TITLE
Update to Symfony 4

### PR DESCRIPTION
**Pull request by pmclain**
### Description
Updates libraries to versions supporting PHP 7.2. This includes the Symfony components and brings composer.lock to a state that allows `composer install` on PHP ~7.1.3 and ~7.2.0

### Fixed Issues (if relevant)
1. magento-engcom/php-7.2-support#4

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,22 +33,24 @@
         "colinmollenhour/cache-backend-file": "1.4",
         "colinmollenhour/cache-backend-redis": "1.10.2",
         "colinmollenhour/credis": "1.6",
-        "colinmollenhour/php-redis-session-abstract": "~1.2.2",
-        "composer/composer": "1.4.1",
-        "magento/composer": "~1.2.0",
+        "colinmollenhour/php-redis-session-abstract": "~1.3.8",
+        "composer/composer": "~1.6.0",
+        "friendsofphp/php-cs-fixer": "~2.10.0",
```
Comment (buskamuza) [inline composer.json]: Why is it moved to `require` from `require-dev`?

File: composer.json [hunk 2]
```diff
@@ -34,22 +33,24 @@
...
         "monolog/monolog": "^1.17",
         "oyejorge/less.php": "~1.7.0",
         "pelago/emogrifier": "^2.0.0",
         "phpseclib/phpseclib": "2.0.*",
         "ramsey/uuid": "3.6.1",
-        "sjparkinson/static-review": "~4.1",
-        "symfony/console": "~2.3, !=2.7.0",
-        "symfony/event-dispatcher": "~2.1",
-        "symfony/process": "~2.1",
+        "sebastian/phpcpd": "~3.0.0",
```
Comment (buskamuza) [inline composer.json]: Why is it moved to `require` from `require-dev`?

File: composer.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
         "sort-packages": true
     },
     "require": {
-        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0",
+        "php": "~7.1.3||~7.2.0",
```
Comment (buskamuza) [inline composer.json:15]: Related https://github.com/magento-engcom/php-7.2-support/pull/30
@pmclain , could you also update the related PR to reflect same PHP versions and target it to the same `upgrade-libs` branch? I'll merge both then. Or if you have time, merge it into this PR.

File: composer.json [hunk 4]
```diff
@@ -34,22 +33,24 @@
         "colinmollenhour/cache-backend-file": "1.4",
         "colinmollenhour/cache-backend-redis": "1.10.2",
         "colinmollenhour/credis": "1.6",
-        "colinmollenhour/php-redis-session-abstract": "~1.2.2",
-        "composer/composer": "1.4.1",
-        "magento/composer": "~1.2.0",
+        "colinmollenhour/php-redis-session-abstract": "~1.3.8",
+        "composer/composer": "~1.6.0",
```
Comment (buskamuza) [inline composer.json:37]: Please update `Magento/Framework/composer.json` with the new version.
Same for `symfony/console` and `symfony/process`

**Comment by buskamuza** [comment]
Confirmed: `composer install` runs w/o errors on both PHP 7.1 and 7.2

**Comment by buskamuza** [comment]
Docs needed:
- http://devdocs.magento.com/guides/v2.3/architecture/arch-basics.html and related articles

@shrielenee, could you please help with this? See updated PHP versions https://github.com/magento-engcom/php-7.2-support/pull/86/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780R15

File: composer.json (hunk 4 above)
Comment (pmclain) [reply composer.json:37]: updated

File: composer.json (hunk 3 above)
Comment (pmclain) [reply composer.json:15]: I've applied the updates from #30 to this PR. I'll close #30

File: composer.json (hunk 1 above)
Comment (pmclain) [reply composer.json]: No reason. Moved back to `require-dev`.

File: composer.json (hunk 2 above)
Comment (pmclain) [reply composer.json]: No reason. Moved back to `require-dev`.

**Comment by pmclain** [comment]
@buskamuza The requested changes have been made. I also updated `.travis.yml`, replacing php 7.0 with 7.2.

**Review by buskamuza — APPROVED** [review]

**Comment by buskamuza** [comment]
Waiting for PRs to internal repositories before this one can be merged.

File: dev/tests/acceptance/tests/functional/Magento/FunctionalTest/AdminNotification/composer.json [hunk 5]
```diff
@@ -12,7 +12,7 @@
     },
     "require": {
         "magento/magento2-functional-testing-framework": "1.0.0",
-        "php": "7.0.2|7.0.4|~7.0.6|~7.1.0"
+        "php": "~7.1.3||~7.2.0"
```
Comment (buskamuza) [inline dev/tests/acceptance/tests/functional/Magento/FunctionalTest/AdminNotification/composer.json:15]: I just noticed that MFTF tests has been also updated. I think it's better to be done in a separate task.
MFTF tests are run separately from the Magento application, so they can still run on its own supported PHP version. In scope of updating MFTF dependencies we should verify that the framework is not broken, which increases scope for this task.

Comment (pmclain) [reply dev/tests/acceptance/tests/functional/Magento/FunctionalTest/AdminNotification/composer.json:15]: Sounds good. I'll back those changes out.

File: dev/tests/functional/composer.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
         "sort-packages": true
     },
     "require": {
-        "php": "7.0.2|~7.0.6|~7.1.0",
+        "php": "~7.1.3||~7.2.0",
```
Comment (buskamuza) [inline dev/tests/functional/composer.json:6]: I propose to skip this one too for now, and make the update in a separate task.

**Pull request merged**